volatility/framework/plugins/linux/check_modules.py

```diff
@@ -0,0 +1,87 @@
+# Volatility
+
+# This file is Copyright 2020 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+"""
+@author:       Matt Tressler
+@contact:      matthewtressler10@gmail.com
+"""
+
+import logging
+from typing import List
+
+from volatility.framework import interfaces, renderers, exceptions, constants, contexts
+from volatility.framework.automagic import linux
+from volatility.framework.configuration import requirements
+from volatility.framework.interfaces import plugins
+from volatility.framework.layers import intel
+from volatility.framework.objects import utility
+from volatility.plugins.linux import lsmod
+from volatility.framework.renderers import format_hints
+
+vollog = logging.getLogger(__name__)
+
+
+class check_modules(plugins.PluginInterface):
+    """Compares module list to sysfs info, if available"""
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.TranslationLayerRequirement(name='primary',
+                                                     description='Memory layer for the kernel',
+                                                     architectures=["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(
+                name="vmlinux", description="Linux kernel symbols"),
+
+            requirements.PluginRequirement(
+                name='lsmod', plugin=lsmod.Lsmod, version=(1, 0, 0))
+        ]
+
+    def get_kset_modules(self, vmlinux):
+
+        try:
+            module_kset = vmlinux.object_from_symbol("module_kset")
+        except exceptions.SymbolError:
+            vollog.error("this command is not supported by this profile")
```

Comment:
Given the check directly after this, it and the return seem a little redundant?  I'd probably just set `module_kset = None` and then let the check below fire (and perhaps raise an Exception, mebbe a RunTimeError?).